tests/test_examples.py

```diff
@@ -1276,3 +1273,20 @@ def test_quadratic_flux_optimization_with_analytic_field():
     # optimizer should zero out field since that's the easiest way
     # to get to Bnorm = 0
     np.testing.assert_allclose(things[0].B0, 0, atol=1e-12)
+
+
+@pytest.mark.unit
+def test_second_stage_optimization():
+    """Test optimizing magnetic field for a fixed axisymmetric equilibrium."""
+    eq = get("DSHAPE")
+    field = ToroidalMagneticField(B0=1, R0=3.5) + VerticalMagneticField(B0=1)
+    objective = ObjectiveFunction(QuadraticFlux(eq=eq, field=field))
+    constraints = FixParameter(field, [{"R0": True}, {}])
+    optimizer = Optimizer("lsq-exact")
+    (field,), _ = optimizer.optimize(
+        things=field, objective=objective, constraints=constraints, verbose=2
+    )
+    # TODO: could make this more robust instead of assuming only vertical field changes
+    np.testing.assert_allclose(field[0].R0, 3.5)
+    np.testing.assert_allclose(field[0].B0, 1)  # toroidal field (does not change)
+    np.testing.assert_allclose(field[1].B0, -0.022, rtol=1e-2)  # vertical field
```

Comment:
make ftol and xtol low or zero and let it hit maxiter, should zero out B verical

Reply:
I tried this and also tried different optimizers, and it still always converges to a finite vertical field of about -0.022 T

Reply:
does the quadratic flux go to zero?

Reply:
I think it's an issue with the `QuadraticFlux` objective. It is correctly computing that the field contribution to the error is 0 when there is no vertical field, but the plasma contribution to the normal field error is non-zero. Minimum working example: 
```
eq = get("DSHAPE")
field = ToroidalMagneticField(B0=1, R0=3.5) + VerticalMagneticField(B0=0)
obj = QuadraticFlux(eq=eq, field=field)
obj.build()
B_plasma = obj._constants["B_plasma"]  # B_plasma*n on boundary surface
print(B_plasma)  # this should be an array of 0s, but it is not!
```

Reply:
ahh ok. If it's finite beta we don't expect the vertical field to go to zero, so that's fine. If we want to pretend its vacuum can just set `vacuum=True` when creating the quadratic flux objective and then im pretty sure it should zero out the vertical field